scripts/record_episodes.py

```diff
@@ -40,379 +27,522 @@
-    dt,
-    max_timesteps,
-    camera_names,
-    dataset_dir,
-    dataset_name,
-    overwrite,
+    max_timesteps: int,
+    dataset_dir: str,
+    dataset_name: str,
+    overwrite: bool,
     torque_base: bool = False,
     gravity_compensation: bool = False,
-):
-    print(f'Dataset name: {dataset_name}')
-
-    node = create_interbotix_global_node('aloha')
-
-    # source of data
-    leader_bot_left = InterbotixManipulatorXS(
-        robot_model='wx250s',
-        robot_name='leader_left',
-        node=node,
-        iterative_update_fk=False,
-    )
-    leader_bot_right = InterbotixManipulatorXS(
-        robot_model='wx250s',
-        robot_name='leader_right',
-        node=node,
-        iterative_update_fk=False,
-    )
+    config: Dict = None
+) -> bool:
+    """
+    Capture one episode of robot teleoperation data and save it to a dataset file.
+
+    :param max_timesteps: Maximum number of timesteps to capture in the episode.
+    :param dataset_dir: Directory where the dataset will be saved.
+    :param dataset_name: Name of the dataset file to create.
+    :param overwrite: If True, overwrite existing dataset file if it exists.
+    :param torque_base: Flag to enable base torque during recording.
+    :param gravity_compensation: Enable gravity compensation on leader robots.
+    :param config: Configuration dictionary containing robot and camera settings.
+    :return: True if data collection is successful; False if data quality is low.
+    """
+    # Determine if the robot has a mobile base and set the control frequency
+    IS_MOBILE = config.get("base", False)
+    DT = 1 / config.get("fps", 50)
 
+    # Initialize the ROS node and robot environment
+    node = create_interbotix_global_node("aloha")
     env = make_real_env(
         node=node,
         setup_robots=False,
         setup_base=IS_MOBILE,
         torque_base=torque_base,
+        config=config,
     )
-
     robot_startup(node)
 
-    # saving dataset
+    # Set up the dataset file path and handle overwrites
     if not os.path.isdir(dataset_dir):
         os.makedirs(dataset_dir)
     dataset_path = os.path.join(dataset_dir, dataset_name)
     if os.path.isfile(dataset_path) and not overwrite:
-        print(f'Dataset already exist at \n{dataset_path}\nHint: set overwrite to True.')
+        print(
+            f"Dataset already exists at {dataset_path}\nHint: Set overwrite to True.")
         exit()
 
-    # move all 4 robots to a starting pose where it is easy to start teleoperation, then wait till
-    # both gripper closed
+    # Move robots to starting position and wait for user to start
     opening_ceremony(
-        leader_bot_left,
-        leader_bot_right,
-        env.follower_bot_left,
-        env.follower_bot_right,
-        gravity_compensation=gravity_compensation,
-    )
+        env.robots, gravity_compensation=gravity_compensation, dt=DT)
 
-    # Data collection
+    # Begin data collection
     ts = env.reset(fake=True)
```

Comment:
Reset Env fake is required while recording episodes. So, I'll  changing the reset function.